pkg/dispatcher/jetstream_dispatcher.go

```diff
@@ -316,7 +321,13 @@ func (s *jetSubscriptionsSupervisor) subscribe(ctx context.Context, channel even
 			s.logger.Debug("dispatch message", zap.String("deadLetter", deadLetter.String()))
 		}
 
-		executionInfo, err := s.dispatcher.DispatchMessage(ctx, message, nil, destination, reply, deadLetter)
+		event := tracing.ConvertNatsMsgToEvent(s.logger, stanMsg)
```

Comment:
`event` could be `nil` after this this call.

Are the following methods ready to handle a `nil` event?

I suspect you might also want to increment a metric and/or a trace in the case the `ConvertNatsMsgToEvent` fails. (You can do that here or in the method, but I do wonder if the ~2 lines of power in that method might just be usefully in-lined here.)

(For example, I'm not sure what `tracing.StartTraceFromMessage` will do with `event == nil`)

Reply:
Yes, start trace from message is ready to accept null event:
```
	sc, ok := parseSpanContext(message)
	if !ok {
		logger.Warn("Cannot parse the spancontext, creating a new span")
		return trace.StartSpan(inCtx, spanName)
	}
```
in case of nil message `trace.StartSpan` is called, `inCtx` will be empty parent context:
```
	if message == nil {
		return trace.SpanContext{}, false
	}
```

Reply:
> I suspect you might also want to increment a metric and/or a trace in the case the ConvertNatsMsgToEvent fails.
I'm not sure what did you mean here. Could you please clarify?
Do you just want to move out logic of handling nil event from `StartTraceFromMessage` into `jetstream_dispatcher.go`?

Reply:
@evankanderson , I refactored code, so it does not return nils anymore, added tests for those cases.

Reply:
> (For example, I'm not sure what tracing.StartTraceFromMessage will do with event == nil)
@evankanderson , I addressed the comment, please review when you have time.